### ISSUE TITLE
Support electric-pair-preserve-balance alike

### ISSUE THREAD

**Issue by denisw** [Fuco1/smartparens#385]
In version 24.4, `electric-pair-mode` gained an option named `electric-pair-preserve-balance`:

```
New electric-pair-preserve-balance enabled by default.
Pairing/skipping only kicks in when that help the balance of
parentheses and quotes, i.e. the buffer should end up at least as
balanced as before.
```

It would be great if smartparens would work the same, that is, if it would not add a pair if the buffer is already unbalanced. So in cases like this (`|` is point):

```
foo = "bar|
... more code ...
```

Typing a string quote character should have the following result (as adding a pair would not improve balance):

```
foo = "bar"|
... more code ...
```

In 24.4's `electric-pair-mode`, this works exactly described above, and this is arguably the expected behavior. It would be nice if `smartparens` acted like that, too.

**Comment by halbtuerke** [comment]
What's the status of this issue? Anything needed to help?

**Comment by Fuco1** [comment]
No progress, and it's still just as difficult :/

**Comment by maplewizard** [comment]
How about this feature now? I really need this.

**Comment by Fuco1** [comment]
I have in works something to handle quote behaviour for 1.10 release.  I'm not sure about the other pairs yet.

**Comment by maplewizard** [comment]
in fact ( [ { still cannot work properly. Could you please check the problem?

**Comment by Fuco1** [comment]
This was fixed for strings in 1.10.  For other pairs one can use strict mode.
